athina_logger/tracing/span.py

```diff
@@ -0,0 +1,313 @@
+import datetime
+from typing import Any, Dict, List, Optional, Union
+from .models import SpanCreateModel
+from .util import remove_none_values
+
+class Span:
+
+    def __init__(
+        self,
+        name: str,
+        start_time: Optional[datetime.datetime] = None,
+        end_time: Optional[datetime.datetime] = None,
+        span_type: str = "span",
+        status: Optional[str] = None,
+        attributes: Optional[dict] = None,
+        input: Optional[dict] = None,
+        output: Optional[dict] = None,
+        duration: Optional[int] = None,
+        version: Optional[str] = None,
+    ):
+        if start_time is None:
+            start_time = datetime.datetime.utcnow()
+        else:
+            start_time = start_time.utcnow()
+        self._span = SpanCreateModel(
+            name=name,
+            span_type=span_type,
+            start_time=start_time.isoformat(),
+            end_time=end_time.isoformat() if end_time else None,
+            status=status,
+            attributes=attributes or {},
+            input=input or {},
+            output=output or {},
+            duration=duration,
+            version=version,
+        )
+        self._children = []
+
+    def __repr__(self):
+        return f"Span(name={self._span.name}, dict={remove_none_values(self.to_dict())}, children={self._children})"
+
+    def to_dict(self):
+        span_dict = self._span.model_dump()
+        span_dict["children"] = [child.to_dict() for child in self._children]
+        return span_dict
+
+    def add_span(
+        self,
+        name: str,
+        start_time: Optional[datetime.datetime] = None,
+        end_time: Optional[datetime.datetime] = None,
+        span_type: str = "span",
+        status: Optional[str] = None,
+        attributes: Optional[dict] = None,
+        input: Optional[dict] = None,
+        output: Optional[dict] = None,
+        duration: Optional[int] = None,
+        version: Optional[str] = None,
+    ):
+        span = Span(
+            name=name,
+            start_time=start_time,
+            end_time=end_time,
+            span_type=span_type,
+            status=status,
+            attributes=attributes or {},
+            input=input or {},
+            output=output or {},
+            duration=duration,
+            version=version,
+        )
+        self._children.append(span)
+        return span
+
+    def add_generation(
```

Comment:
Similarly here, it should be able to take just generation as a parameter or we provide another method for that